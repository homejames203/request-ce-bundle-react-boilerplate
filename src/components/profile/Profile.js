import React from 'react';
import { Seq, Set } from 'immutable';
import DocumentTitle from 'react-document-title';

const INPUT_PROPS =
  Set(['name', 'onBlur', 'onChange', 'onDragStart', 'onDrop', 'onFocus', 'value']);

// Redux form passes in several props that are not valid for react 'input'
// elements. React has since started to print warnings about these props so the
// helper below is used to filter out props that react will complain about.
const filterProps = props =>
  Seq(props).filter((value, key) => INPUT_PROPS.contains(key)).toJS();

const Field = props =>
  <div>
    <input
      {...filterProps(props.controls)}
      type={props.type}
      value={props.controls.value || ''}
      disabled={props.disabled}
    />
  </div>;

export const Profile = ({ handleSubmit, fields, loading, apiErrors }) =>
  <DocumentTitle title="Profile">
    <div>
      {loading && <p>Loading profile information...</p>}
      {apiErrors.map(error => <p>{error}</p>)}
      <form onSubmit={handleSubmit} noValidate>
        <div>
          <label htmlFor="username">Username</label>
          <Field type="text" id="username" controls={fields.username} disabled={loading} />
        </div>
        <div>
          <label htmlFor="displayName">Display Name</label>
          <Field type="text" id="displayName" controls={fields.displayName} disabled={loading} />
        </div>
        <div>
          <label htmlFor="email">Email</label>
          <Field type="email" id="email" controls={fields.email} disabled={loading} />
        </div>
        <button type="submit" disabled={loading}>Submit</button>
      </form>
    </div>
  </DocumentTitle>;
