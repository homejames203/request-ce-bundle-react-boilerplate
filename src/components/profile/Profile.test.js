import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Profile } from './Profile';

const props = {
  handleSubmit: jest.fn(),
  fields: {
    username: {},
    displayName: {},
    email: {},
  },
  loading: false,
  apiErrors: [],
};

describe('<Profile />', () => {
  test('verify snapshot', () => {
    const component = shallow(<Profile {...props} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
