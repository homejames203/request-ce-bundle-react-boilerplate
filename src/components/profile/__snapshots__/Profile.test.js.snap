// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> verify snapshot 1`] = `
<SideEffect(DocumentTitle)
  title="Profile"
>
  <div>
    <form
      noValidate={true}
      onSubmit={[Function]}
    >
      <div>
        <label
          htmlFor="username"
        >
          Username
        </label>
        <Field
          controls={Object {}}
          disabled={false}
          id="username"
          type="text"
        />
      </div>
      <div>
        <label
          htmlFor="displayName"
        >
          Display Name
        </label>
        <Field
          controls={Object {}}
          disabled={false}
          id="displayName"
          type="text"
        />
      </div>
      <div>
        <label
          htmlFor="email"
        >
          Email
        </label>
        <Field
          controls={Object {}}
          disabled={false}
          id="email"
          type="email"
        />
      </div>
      <button
        disabled={false}
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</SideEffect(DocumentTitle)>
`;
